primary_beam_correction.py

```diff
@@ -132,70 +123,62 @@ def standard_deviation(data):
     while diff > 0:
         i += 1
         if i > 50:
-            return sd
+            return 1/(sd)**2
         old_sd = sd
         old_len = len(data)
         cut = np.abs(data - med) < 5.0 * sd
         if np.all(~cut):
-            return sd
+            return 1/(sd)**2
         data = data[cut]
-        sd = mad(data)[1]
+        med, sd = mad(data)
         diff = old_len - len(data)
         if sd == 0.0:
             return old_sd
     return 1/(sd)**2
 
 
-def weigted_average(arr, weights):
-    '''
+def weighted_average(arr, weights):
+    """
     Computing weighted average of all the frequency planes.
-    '''
+    """
     wt_average = np.average(arr, weights=weights, axis=0)
```

Comment:
I still think there is a problem here that if one plane of the input image is completely blank then the whole output is completely blank. Is there somewhere else in the code that deals with this that I missed?

Reply:
I think we need a quick discussion of this over hangouts. My initial code was dealing with that exception but you suggested that I remove it.

Reply:
```
def weighted_average(arr, weights, mask):
    """
    Computing weighted average of all the frequency planes.
    """
    masked_arr = np.ma.MaskedArray(arr, mask=mask)
    wt_average = np.average(masked_arr, weights=weights, axis=0)
    wt_average.data[mask] = np.nan
    return wt_average.data
```

Reply:
Done